docs/src/contributing/releasing.md

```diff
@@ -0,0 +1,156 @@
+# Releasing Neptune Core
+
+This section describes the steps to publish a new version of Neptune Core, and to release & distribute its binary artifacts.
+
+## Pre-requisites
+
+The following tools are used to ensure a high quality release.
+
+- [cargo-binstall](https://github.com/cargo-bins/cargo-binstall) – Faster installation of the needed tools (optional)
+- [cargo-semver-checks](https://github.com/obi1kenobi/cargo-semver-checks/) – Scans the crate for [semver](https://semver.org/) violations
+- [git cliff](https://git-cliff.org/docs/) – Simplifies changelog creation
+- [cargo-release](https://github.com/crate-ci/cargo-release) – Simplifies simultaneous publication of multiple crates
+- [dist](https://opensource.axo.dev/cargo-dist/book/introduction.html) – Creates installers and publishes them in a [GitHub release](https://github.com/Neptune-Crypto/neptune-core/releases)
+
+Use the following commands to install the needed tools.
+If you decide against using `cargo binstall`, it's generally possible to just `cargo install` instead.
+Some tools might require `cargo install --locked`.
+
+```sh
+cargo install cargo-binstall
+cargo binstall cargo-semver-checks
+cargo binstall git-cliff
+cargo binstall cargo-release
+cargo binstall cargo-dist
+```
+
+## Release Process Checklist
+
+Not every step of the release process is (or should be) fully automated.
+An example of a semi-automated step is changelog generation.
+Tools like `git cliff` help, but a manual edit is necessary to reduce noise and achieve the polish appreciated by readers of the changelog.
+An example of a fully automated step is assembly and distribution of binaries by `dist`.
+
+### Set Working Directory to Workspace Root
+
+Unless indicated otherwise, the current working directory is assumed to be the workspace root.
+
+```sh
+cd /path/to/neptune-core
+```
+
+### Bump Version
+
+Bump the version in `Cargo.toml` [as appropriate](https://doc.rust-lang.org/cargo/reference/semver.html).
+
+### Confirm Version Bump as Semantic
+
+> ℹ️ Because binaries cannot be used as a dependency, this step is only relevant if Neptune Core has library targets.
```

Comment:
I think library targets is something we should support, but probably not urgently.

I'm thinking of external developers of external applications who want to do something trivial like parsing one or two types of messages or verifying a proof. They should be able to do so with minimal hassle, and I think having an importable library helps in that regard. Otherwise they must *a)* run a neptune node or *b)* copy code, is that right? However, I'm not sure what the next step would be towards making this codebase more library-like, so please comment if you know more than I do.

Reply:
A cargo package supports [up to one library target](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#library) among its potentially many binary targets. It should be possible to split the binary `neptune-core` into a library and surround it with a binary wrapper without having to publish multiple crates. I don't know the architecture well enough to be able to suggest a natural fault line; alsmost certainly, anything handling command line arguments belongs into the binary, not the library.